scripts/globals/events/egg_hunt_egg-stravaganza.lua

```diff
@@ -0,0 +1,898 @@
+    REWARD2 = 1,  -- Egg-straordinary! Here's your special prize, kupo! But how did you figure out the secret combination...?
+    REWARD3 = 2,  -- How in-eggs-plicably astounding! You're truly a hard-boiled detective of the highest degree, kupo! Here's your prize, and don't forget to enjoy the rest of the festivities.
+    WAITING = 3,  -- Hey, don't get all egg-cited! You can only claim one prize a day, kupo!
+    HINTING = 4,  -- These initial eggs are really egg-spensive, kupo! If you want one I'm going to need a little more incentive. No egg-ceptions!
+    TRADE1  = 5,  -- Eggs-actly what I wanted! Here's your reward, kupo!
+    TRADE2  = 6,  -- Ku-kupo! Such an egg-stravagant item, for little old me? Here's a whole basketful of initial eggs for you, kupo!
+    TRADE3  = 7,  -- My eyes! My eyes! Get out of here, and take those egg-ceptionally nasty things with you, kupo!
+    TRADE4  = 8,  -- Kupo... Kupo! Please! No more, kupo! Here, take these and get out of here, kupo!
+    DELAY   = 9,  -- Hey, I can only trade with you once a day. We wouldn't want to make the egg hunt over easy, now, kupo!
+    HELP    = 10, -- Hey, you look new here, kupo! If you need anything, just let me know!
+    REWARD4 = 11, -- Heh heh heh... I have something egg-specially egg-citing in store for you, kupo!
+}
+
+------------------------------------
+-- Helpers
+------------------------------------
+
+xi.events.egg_hunt.charToEgg = function(char)
+    local letter = string.byte(string.lower(char)) - 97
+    return xi.items.A_EGG + letter
+end
+
+xi.events.egg_hunt.eggInList = function(eggs, egg)
+    for j = 1, #eggs do
+        if eggs[j] and eggs[j][1] == egg then
+            return j
+        end
+    end
+
+    return -1
+end
+
+xi.events.egg_hunt.stringToEggs = function(text)
+    local str = string.lower(text)
+    local eggs = {}
+
+    for i = 1, #str do
+        local letter = string.byte(string.sub(str, i, i))
+        local egg = xi.items.A_EGG + (letter - 97)
+        local pos = xi.events.egg_hunt.eggInList(eggs, egg)
+
+        if pos > -1 then
+            eggs[pos][2] = eggs[pos][2] + 1
+        else
+            table.insert(eggs, { egg, 1 })
+        end
+    end
+
+    return eggs
+end
+
+local hasFirstThree = function(player)
+    return (
+        player:getCharVar(settings.VAR.FIRST_THREE) == 1 or
+        player:hasItem(xi.items.EGG_HELM)
+    )
+end
+
+------------------------------------
+-- Combos
+------------------------------------
+
+local minorRewards =
+{
+    xi.items.CHOCOBO_TICKET,
+    xi.items.RED_DROP,
+    xi.items.YELLOW_DROP,
+    xi.items.BLUE_DROP,
+    xi.items.GREEN_DROP,
+    xi.items.CLEAR_DROP,
+    xi.items.PURPLE_DROP,
+    xi.items.WHITE_DROP,
+    xi.items.BLACK_DROP,
+}
+
+local minorReward = function()
+    return minorRewards[math.random(#minorRewards)]
+end
+
+local tradeIn = function(player, npc, trade, zoneID)
+    local reward = {}
+
+    if
```

Comment:
The three below conditions are exclusive, since they depend on tradeHasExactly().  There's not really a reason to have separate blocks for these outside of readability (which I definitely understand).  As a future effort, we might want to table up conditions and iterate over that.